src/wallet/wallet.cpp

```diff
@@ -212,6 +212,7 @@ std::shared_ptr<CWallet> LoadWalletInternal(interfaces::Chain& chain, const std:
         std::shared_ptr<CWallet> wallet = CWallet::Create(chain, name, std::move(database), options.create_flags, error, warnings);
         if (!wallet) {
             error = Untranslated("Wallet loading failed.") + Untranslated(" ") + error;
+            status = DatabaseStatus::FAILED_CREATE;
```

Comment:
In commit "bugfix: make LoadWallet assigns status always" (f32ddd81267cfca9f3c07006cef7c80f79a48527)

Better to use `DatabaseStatus::FAILED_LOAD` here because the error happens when loading an existing wallet not creating a new one. (`CWallet::Create` name refers to creating a `CWallet` object not creating a new database).